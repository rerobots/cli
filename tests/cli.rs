// SCL <scott@rerobots.net>
// Copyright (C) 2021 rerobots, Inc.

use assert_cmd::Command;


#[test]
fn prints_version() {
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("-V").assert();
    assert
        .stdout(format!("{}\n", env!("CARGO_PKG_VERSION")))
        .success();
}


#[test]
fn prints_help() {
    let mut cmd = Command::cargo_bin("rerobots").unwrap();
    let assert = cmd.arg("help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!(String::from_utf8(output.stdout).unwrap());
}
