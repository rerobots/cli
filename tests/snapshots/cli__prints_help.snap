---
source: tests/cli.rs
assertion_line: 24
expression: "String::from_utf8(output.stdout).unwrap()"
---
rerobots API command-line client 

USAGE:
    rerobots [FLAGS] [OPTIONS] [SUBCOMMAND]

FLAGS:
    -n               assume "no" for any questions required to execute the
                     command; this can prevent destructive actions, e.g.,
                     overwriting a local file
    -y               assume "yes" for any questions required to execute the
                     command; otherwise, interactive prompts will appear to
                     confirm actions as needed
    -h, --help       Prints help information
    -v, --verbose    Increases verboseness level of logs; ignored if RUST_LOG is
                     defined
    -V, --version    Prints version number and exits

OPTIONS:
    -t <FILE>                plaintext file containing API token; with this
                             flag, the REROBOTS_API_TOKEN environment variable
                             is ignored
        --format <FORMAT>    output formatting; options: YAML , JSON

SUBCOMMANDS:
    get-ssh-key    Get secret key for SSH access to instance
    help           Prints this message or the help of the given
                   subcommand(s)
    info           Print summary about instance
    isready        Indicate whether instance is ready with exit code
    launch         Launch instance from specified workspace deployment or
                   type
    list           List all instances by this user
    search         Search for matching deployments. empty query implies show
                   all existing workspace deployments
    ssh            Connect to instance host via ssh
    terminate      Terminate instance
    token          Get information about an API token
    version        Prints version number and exits
    wdinfo         Print summary about workspace deployment

